api.go

```diff
@@ -0,0 +1,209 @@
+/*
+ * Copyright 2025 Hypermode Inc.
+ * Licensed under the terms of the Apache License, Version 2.0
+ * See the LICENSE file that accompanied this code for further details.
+ *
+ * SPDX-FileCopyrightText: 2025 Hypermode Inc. <hello@hypermode.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package modusdb
+
+import (
+	"fmt"
+
+	"github.com/dgraph-io/dgraph/v24/dql"
+	"github.com/dgraph-io/dgraph/v24/schema"
+	"github.com/hypermodeinc/modusdb/api/apiutils"
+	"github.com/hypermodeinc/modusdb/api/structreflect"
+)
+
+func Create[T any](db *DB, object T, ns ...uint64) (uint64, T, error) {
+	db.mutex.Lock()
+	defer db.mutex.Unlock()
+	if len(ns) > 1 {
+		return 0, object, fmt.Errorf("only one namespace is allowed")
+	}
+	ctx, n, err := getDefaultNamespace(db, ns...)
+	if err != nil {
+		return 0, object, err
+	}
+
+	gid, err := db.z.nextUID()
+	if err != nil {
+		return 0, object, err
+	}
+
+	dms := make([]*dql.Mutation, 0)
+	sch := &schema.ParsedSchema{}
+	err = generateSetDqlMutationsAndSchema[T](ctx, n, object, gid, &dms, sch)
+	if err != nil {
+		return 0, object, err
+	}
+
+	err = n.alterSchemaWithParsed(ctx, sch)
+	if err != nil {
+		return 0, object, err
+	}
+
+	err = applyDqlMutations(ctx, db, dms)
+	if err != nil {
+		return 0, object, err
+	}
+
+	return getByGid[T](ctx, n, gid)
```

Comment:
I'm so/so on whether these create/upsert functions should always do a get at the end.  The user could just do that themselves.  And in the case of multiple inserts, they may just throw away the result so it's extraneous work that affects insert perf.